pkg/apis/triggers/v1alpha1/event_listener_validation_test.go

```diff
@@ -23,6 +23,23 @@ func Test_EventListenerValidate_error(t *testing.T) {
 			bldr.EventListenerSpec(
 				bldr.EventListenerTrigger("tb", "dne", "v1alpha1"))),
 		raiseErr: true,
+	}, {
+		name: "Both Binding and Bindings Present",
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Bindings:          []*v1alpha1.EventListenerBinding{{Name: "tb"}},
+					DeprecatedBinding: &v1alpha1.EventListenerBinding{Name: "bar"},
+					Template:          v1alpha1.EventListenerTemplate{Name: "tt"},
+				}},
+			},
+		},
+
+		raiseErr: true,
```

Comment:
in general i find that the tests are easier to read and maintain if we separate the error cases from the success cases into completely different functions

Reply:
Agree in general but I feel like these tests are simple enough that a single function should work too. Unless you feel strongly about this, I'd prefer keeping this in now and maybe splitting it later if the tests become more complex.

Reply:
I'm afraid I do feel strongly! Besides the additional complexity i think its much easier to see what cases are covered if you can look at a list of error cases completely separately from a list of success cases.

Looking at the body of the text i can see that a lot of setup happens that would theoretically need to be duplicated in both sets of tests, is that why you want to keep it in one test?

(also i think this is total scope creep since you are just updating this test in this PR tho so im happy to just let it go)

Reply:
I don't feel too strongly about this but since I was just adding a test case, wanted to keep the scope limited. I'll make the change in a follow up